src/Build/BackEnd/BuildManager/BuildRequestData.cs

```diff
@@ -75,6 +75,11 @@ public enum BuildRequestDataFlags
         /// This is especially useful during a restore since some imports might come from packages that haven't been restored yet.
         /// </summary>
         IgnoreMissingEmptyAndInvalidImports = 1 << 6,
+
+        /// <summary>
+        /// Turns on logic for graph based isolated project builds
+        /// </summary>
+        IsolatedProjectBuild = 1 << 7
```

Comment:
Should this be `ExecuteExportedTargets` instead? I bet we're going to need `IsolatedProjectBuild` as we do more things differently for isolated project builds.

Reply:
Or instead of using a master switch we use n fine grained switches which get turned on by the Scheduler?

Reply:
We'll eventually need it for enforcement. Would this be where we'd read it from?

Reply:
I guess there's two ways of toggling these for builds:
- globally via build parameters. This means that the entire build (all builds submitted in one begin / end build session) should go into `isolated build project` mode, or `execute export targets` mode
- locally for each project build request. This means that we can control these things per each build.

Would we ever want to mix these things? If not, then maybe build parameters would be better. In which case it should be internal so we don't clutter the public APIs.

Reply:
We'll never need to mix and match. The build is either an isolated build or not.

Reply:
(So BuildParameters is probably the right thing to use here if possible)